server/controllers/auth.js

```diff
@@ -0,0 +1,113 @@
+            if (err){
+                const error = new ErrorResponse("Error in database finding user.", 500);
+                return next(error);            
+            } else if (userExistance){
+                const error = new ErrorResponse("User Already Exists.", 403);
+                return next(error);
+            } else {
+                const user = new User({
+                      email,
+                      password,
+                      name,
+                  })
+                user.save((err, data) => {
+                    if (err){
+                        const error = new ErrorResponse("Error in database saving user.", 401);
+                        return next(error);
+                    } else if(data) {
+                        data.password = undefined
+                        token = accessToken(data.id)
+                        return res.json({ 
+                            user: data,
+                            token: token
+                        })
+                    }
+                })
+            }        
+        }).exec();
+    } catch (err) {
+        return next(err)
+    }
+}
+
+/* @desc: logInController takes a request and validates email and password.
+ * As well, determines if user already exists via email and if one does exists
+ * the retrieved user document's password is compared to the request's password. 
+ * If the passwords match the response object is modified with a JSON object that contains
+ * a JWT and the retrieved user document without a password. 
+ * @param: req, res, next
+ * @returns: none
+*/
+module.exports.logInController = async (req, res, next) => {
+    const err = validationResult(req).formatWith(errorFormater);
+    if (!err.isEmpty()) {
+        const error = new ErrorResponse(err.array(), 401);
+        return next(error);
+    } 
+
+    const { email, password } = req.body;
+    try {
+        const user = await User.findOne({
+            email
+        })
+        if (!user) {
+            const error = new ErrorResponse("User does not exist.", 401);
+            return next(error);
+        }
+        user.comparePassword(password, function(err, isMatch){
+            if (err){
+                const error = new ErrorResponse("Unable to compare password.", 500);
+                return next(error);
+            } else if (isMatch){
+                token = accessToken(user.id)
+                user.password = undefined
+                res.json({
+                    user,
+                    token,
+                })
+            } else {
+                const error = new ErrorResponse("Incorrect password.", 403);
+                return next(error);            }
+        })
+
+    } catch (err) {
+        return next(err)
+    }
+
+}
+
+/* @desc: A function to test isAuthorized function is verifying JWT correctly. 
+ * @param: req, res, next
+ * @returns: none
+*/
+module.exports.tokenVerifier = async (req, res, next) => {
```

Comment:
This function does not check the toke inside the function. isAuthorized does check the token, you can find that function inside the authorization file under middlewares. isAuthorized is called before this function in the route /api/v1/auth/tokenVerifier in the authRoutes file. This function, tokenVerifier, just verifies that isAuthorized is working. is Authorized is suppose to verify the token and then assign the user to req.user so that tokenVerifier can show that isAuthorized works by placing req.user in the response.

Reply:
Ok. I got it